backtrace-library/src/main/java/backtraceio/library/common/ApplicationMetadataCache.java

```diff
@@ -0,0 +1,102 @@
+package backtraceio.library.common;
+
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+
+import backtraceio.library.logger.BacktraceLogger;
+
+public class ApplicationMetadataCache {
+
+    private static final transient String LOG_TAG = ApplicationMetadataCache.class.getSimpleName();
+
+    private static volatile ApplicationMetadataCache instance;
+
+    /**
+     * Cached application name
+     */
+    private String applicationName;
+
+    /**
+     * Cached application version
+     */
+    private String applicationVersion;
+
+    /**
+     * Cached package name
+     */
+    private String packageName;
+
+    /**
+     * Returns current application cache. This instance is a singleton since we can only operate
+     * in a single application scope.
+     *
+     * @param context Application context
+     * @return Application metadata cache
+     */
+    public static ApplicationMetadataCache getInstance(Context context) {
```

Comment:
please move this method below constructor

Reply:
btw maybe we can split this method into `init` and `getInstance` to init something in `get` method

Reply:
re second advice - I would prefer to stay as it is